src/tokens/TokenImport.tsx

```diff
@@ -63,13 +68,41 @@ export default function TokenImportScreen(_: Props) {
       if (supportedTokens[tokenId]) {
         setAddressState(AddressState.AlreadySupported)
       } else {
-        setAddressState(AddressState.Valid)
+        validateCheck(address)
       }
     } else {
       setAddressState(AddressState.Invalid)
     }
   }
 
+  const validateCheck = async (address: Address) => {
+    setAddressState(AddressState.FetchingContract)
+
+    const contract = getContract({
+      abi: erc20.abi,
+      address,
+      publicClient: publicClient[networkIdToNetwork[networkId]],
+    })
+
+    const state = await Promise.race<AddressState>([
+      new Promise((resolve) => {
+        Promise.all([
+          contract.read.symbol(),
```

Comment:
in the viem example docs i see the syntax
```
publicClient.readContract({
    abi: erc20.abi,
      address,
      functionName: 'symbol',
 }),
```
is there any advantage in your approach over theirs? it seems like what is here we fetch the contract and then read it, but `readContract` does this in one step? i'm not sure, just curious

Reply:
Contract instance (`getContract(..)` returned value) abstracts calls to `readContract` as the instance knows the ABI and address. If we need to read multiple public variables (or functions with view or pure state mutability), the code will be more concise with the contract instance. Otherwise they're the same.

[Documentation](https://viem.sh/docs/contract/getContract.html#:~:text=Contract%20Instances%20can%20make%20it%20easier%20to%20work%20with%20contracts%20if%20you%20don%27t%20want%20to%20pass%20the%20abi%20and%20address%20properties)